com.unity.shadergraph/Editor/Drawing/Views/MaterialGraphView.cs

```diff
@@ -841,25 +841,38 @@ void ConvertToInlineNode(DropdownMenuAction action)
 
         void DuplicateSelection()
         {
-            graph.owner.RegisterCompleteObjectUndo("Duplicate Blackboard Property");
+            graph.owner.RegisterCompleteObjectUndo("Duplicate Blackboard Selection");
 
             List<ShaderInput> selectedProperties = new List<ShaderInput>();
+            List<CategoryData> selectedCategories = new List<CategoryData>();
+
+            foreach (var selectable in selection.ToList())
```

Comment:
Why `ToList()`? Can't you just enumerate the selection?

Reply:
Doing it directly without having a `ToList()` is a problem further in the loop, because we're also trying to modify the selection by removing items from it

Not having the `ToList()`, causes an `InvalidOperationException`

Reply:
But, Chris suggested a better way to do a list enumeration while removing items below in one of his comments, and I'm going to run with that using the actual object indices and `RemoveAll` instead to avoid the list GC here